test/functional/fixtures/regression/gh-1956/pages/index.html

```diff
@@ -0,0 +1,105 @@
+<html>
+<head><title>Edit test</title></head>
+<style>
+    h2 {
+        margin-top: 20px;
+    }
+    div {
+        border: 1px solid black;
+        min-height: 10px;
+    }
+</style>
+<script type="text/javascript">
+    function onTextInput (event) {
+        var needPreventDefault = true;
+
+        if (isTargetElement(event.target, 'contentEditableWithModify'))
+            changeNodeValueOnTextInput(event);
+
+        if (isTargetElement(event.target, 'contentEditableWithReplace')) {
+            replaceEditableElementOnTextInput(event);
+
+            needPreventDefault = false;
+        }
+
+        if(needPreventDefault)
+            event.preventDefault();
+    }
+
+    function changeNodeValueOnTextInput(event) {
+        document.getElementById('contentEditableWithModify').childNodes[0].childNodes[0].nodeValue += event.data;
+    }
+
+    function replaceEditableElementOnTextInput (event) {
+        if (window.preventNextElementReplacement)
+            return;
+
+        window.preventNextElementReplacement = true;
+
+        var div          = document.getElementById('contentEditableWithReplace');
+        var paragraph    = div.childNodes[0];
+        var textNode     = paragraph.childNodes[0];
+        var newParagraph = document.createElement("P");
+
+        newParagraph.innerHTML = 'X';
+
+        div.removeChild(paragraph);
+        div.appendChild(newParagraph);
+        newParagraph.focus();
+    }
+
+    function onInput (event) {
+        if (!isTargetElement(event.target, 'contentEditableWithReplace'))
+            throw new Error('Input event has raised');
+    }
+
+    function isTargetElement(el, id) {
+        while (el) {
```

Comment:
we can store required elements (such as `const contentEditableWithModify = document.getElementById('contentEditableWithModify')`) and use DOM element `contains` method